src/bioregistry/data/bioregistry.json

```diff
@@ -34871,6 +34871,18 @@
     },
     "uri_format": "http://purl.bioontology.org/ontology/OPB?conceptid=http%3A%2F%2Fbhi.washington.edu%2FOPB%23OPB_$1"
   },
+  "openalex": {
+    "contributor": {
+      "name": "Daniel Himmelstein",
+      "orcid": "0000-0002-3012-7446"
+    },
+    "description": "OpenAlex is a fully open catalog of the global research system that describes scholarly entities and how those entities are connected to each other.",
+    "example": "W2741809807",
+    "homepage": "https://openalex.org/",
+    "name": "OpenAlex",
+    "pattern": "^[WAVIC]\\d+$",
```

Comment:
@hpiwowar is this the best regex for validating an OpenAlex id? Are there additional constrains we could place on the number of digits?

Reply:
From https://docs.openalex.org/entity-objects#the-openalex-id:

> The first letter in the unique part of the ID tells you what kind of entity you've got. In the example above, the unique part of the ID is `W2741809807`; the `W` at the front tells us that this is a `Work`. The other possible first letters are `A` (`Author`), `V` (`Venue`), `I` (`Institution`), and `C` (`Concept`). The IDs are not case-sensitive, so `w2741809807` is just as valid as `W2741809807`.

So we should probably allow lowercase.

Reply:
Based on https://github.com/ourresearch/openalex-guts/blob/main/util.py#L199 , it looks like they're only validating upper case, but maybe there are string transformations applied before. I think it's better to be a bit more pedantic in the bioregistry, so let's keep just upper.

Reply:
Versioned link https://github.com/ourresearch/openalex-guts/blob/a4f9f8b3692d7e73cd6620273442429275fad498/util.py#L199

Reply:
Can we add the two digit minimum `\d{2,}`?

Reply:
Done in https://github.com/biopragmatics/bioregistry/pull/281/commits/1d235a12d343f92c63d81a5ce11807057caf5eee